edb/server/connpool/pool.py

```diff
@@ -108,7 +109,7 @@ class Block(typing.Generic[C]):
     conn_acquired_num: int
     conn_waiters_num: int
     conn_waiters: typing.Deque[asyncio.Future[None]]
-    conn_queue: typing.Deque[C]
+    conn_stack: typing.Deque[typing.Tuple[float, C]]
```

Comment:
Hm, maybe we put the timestamp into `ConnectionState`?

Reply:
Yeah I think that'll work - it's just a little weird to have:

```python
    def try_steal(
        self, only_older_than: float = float("inf")
    ) -> typing.Optional[C]:
        if (
            self.conn_stack
            and self.conns[self.conn_stack[0]].in_stack_since < only_older_than
        ):
            conn = self.conn_stack.popleft()
            assert conn in self.conns  # <--- we don't need this any more
            return conn
        return None
```

But I guess it runs at about the same speed, so I'm basically fine with either.

Reply:
This is easier to read:

```python
    def try_steal(
        self, only_older_than: typing.Optional[float] = None
    ) -> typing.Optional[C]:

        if not self.conn_stack:
            return None
        
        conn = self.conn_stack.popleft()
        assert conn in self.conns
        
        if only_older_than is not None:
            if self.conns[conn].in_stack_since > only_older_than:
                self.conn_stack.appendleft(conn)
                return None
            
        return conn
```

Reply:
This is a bit more optimal, imo:

```python
    def try_steal(
        self, only_older_than: typing.Optional[float] = None
    ) -> typing.Optional[C]:

        if not self.conn_stack:
            return None
        
        if only_older_than is not None:
            top_conn = self.conn_stack[0]
            if self.conns[top_conn].in_stack_since > only_older_than:
                return None            
            
        return self.conn_stack.popleft()
```